test/timeslot/delete-test.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const expect = require('chai').expect;
+const request = require('superagent');
+const Promise = require('bluebird');
+
+const Spot = require('../../model/spot.js');
+const Lot = require('../../model/lot.js');
+const User = require('../../model/user.js');
+const Timeslot = require('../../model/timeslot.js');
+
+require('../../server.js');
+
+const url = `http://localhost:${process.env.PORT}`;
+
+const exampleUser = {
+  name: 'exampleuser',
+  password: '12345',
+  email: 'example@test.com'
+};
+
+const exampleLot = {
+  name: 'example lot',
+  description: 'example lot description',
+  address: 'example address',
+};
+
+const exampleSpot = {
+  name: 'example spot',
+  description: 'example spot description',
+};
+
+const exampleTimeslot = {
+  startTime: Date.now,
+  endTime: Date.now
+};
+
+describe('Timeslot Delete Route', function () {
+  describe('DELETE: /api/lot/:lotID/spot/:spotID/timeslot/:id', function () {
+    before(done => {
+      new User(exampleUser)
+        .generatePasswordHash(exampleUser.password)
+        .then( user => {
+          return user.save();
+        })
+        .then(user => {
+          this.tempUser = user;
+          return user.generateToken();
+        })
+        .then(token => {
+          this.tempToken = token;
+        })
+        .then(() => {
+          exampleLot.userID = this.tempUser._id.toString();
+          new Lot(exampleLot).save()
+            .then( lot => {
+              this.tempLot = lot;
+              done();
+            })
+            .catch(done);
+        });
+
+      before(done => {
+        new Spot(exampleSpot).save()
+          .then(spot => {
+            this.tempSpot = spot;
+            done();
+          })
+          .catch(done);
+      });
+
+      after(done => {
+        Promise.all([
+          User.remove({}),
+          Lot.remove({}),
+          Spot.remove({}),
+          Timeslot.remove({})
```

Comment:
You don't need to call `Timeslot.remove()` because your delete route will delete the test timeslot you created in your before block.